File: frontend/src/app/admin/services/admin-dashboard.service.ts
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';

import {
  DashboardStats,
  OrderItem,
  PopularProduct,
} from '../models/admin-dashboard.models';
import { ApiHttpClient } from '../../core/http/http-client';

@Injectable({
  providedIn: 'root',
})
export class AdminDashboardService {
  private readonly api = inject(ApiHttpClient);

  getStats(): Observable<DashboardStats> {
    return this.api.get<DashboardStats>('/admin/dashboard/stats');
  }

  getRecentOrders(): Observable<OrderItem[]> {
    return this.api.get<OrderItem[]>('/admin/dashboard/orders/recent');
  }

  getPopularProducts(): Observable<PopularProduct[]> {
    return this.api.get<PopularProduct[]>('/admin/dashboard/products/popular');
  }
}

File: backend/Ecommarce.Api/Services/AdminDashboardService.cs
using Ecommarce.Api.Models;
using Ecommarce.Api.Repositories;

namespace Ecommarce.Api.Services;

public sealed class AdminDashboardService : IAdminDashboardService
{
    private readonly IAdminRepository _repository;

    public AdminDashboardService(IAdminRepository repository)
    {
        _repository = repository;
    }

    public DashboardStats GetDashboardStats() => _repository.GetDashboardStats();

    public List<OrderItem> GetRecentOrders() => _repository.GetRecentOrders();

    public List<PopularProduct> GetPopularProducts() => _repository.GetPopularProducts();
}
